bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericDiscoveryParticipant.java

```diff
@@ -0,0 +1,107 @@
+
+    private final Logger logger = LoggerFactory.getLogger(GenericDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Set.of(GenericBindingConstants.THING_TYPE_GENERIC);
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(BluetoothDiscoveryDevice device) {
+        ThingUID thingUID = getThingUID(device);
+        if (thingUID == null) {
+            // the thingUID will never be null in practice but this makes the null checker happy
+            return null;
+        }
+        String label = "Generic Connectable Bluetooth Device";
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(BluetoothBindingConstants.CONFIGURATION_ADDRESS, device.getAddress().toString());
+        Integer txPower = device.getTxPower();
+        if (txPower != null && txPower > 0) {
+            properties.put(BluetoothBindingConstants.PROPERTY_TXPOWER, Integer.toString(txPower));
+        }
+        String manufacturer = BluetoothCompanyIdentifiers.get(device.getManufacturerId());
+        if (manufacturer == null) {
+            logger.debug("Unknown manufacturer Id ({}) found on bluetooth device.", device.getManufacturerId());
+        } else {
+            properties.put(Thing.PROPERTY_VENDOR, manufacturer);
+            label += " (" + manufacturer + ")";
+        }
+
+        addPropertyIfPresent(properties, Thing.PROPERTY_MODEL_ID, device.getModel());
+        addPropertyIfPresent(properties, Thing.PROPERTY_SERIAL_NUMBER, device.getSerialNumber());
+        addPropertyIfPresent(properties, Thing.PROPERTY_HARDWARE_VERSION, device.getHardwareRevision());
+        addPropertyIfPresent(properties, Thing.PROPERTY_FIRMWARE_VERSION, device.getFirmwareRevision());
+        addPropertyIfPresent(properties, BluetoothBindingConstants.PROPERTY_SOFTWARE_VERSION,
+                device.getSoftwareRevision());
+
+        return DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                .withRepresentationProperty(BluetoothBindingConstants.CONFIGURATION_ADDRESS)
+                .withBridge(device.getAdapter().getUID()).withLabel(label).build();
+    }
+
+    private static void addPropertyIfPresent(Map<String, Object> properties, String key, @Nullable Object value) {
+        if (value != null) {
+            properties.put(key, value);
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(BluetoothDiscoveryDevice device) {
+        return new ThingUID(GenericBindingConstants.THING_TYPE_GENERIC, device.getAdapter().getUID(),
+                device.getAddress().toString().toLowerCase().replace(":", ""));
+    }
+
+    @Override
+    public boolean requiresConnection(BluetoothDiscoveryDevice device) {
+        return true;
+    }
+
+    @Override
+    public int order() {
+        // we want to go last
+        return Integer.MAX_VALUE;
```

Comment:
Generic bluetooth devices were always intended to be the fallback. Originally this binding was integrated into the bluetooth binding code and used as the fallback during discovery, but I separated it into its own binding so I had to add this `order` method to the discovery participant class so that the generic bluetooth devices could still act as the fallback. If there is ever a time that something else should be the fallback (I can't imagine what) then we can change the  `order` value of generic devices.